staging/src/k8s.io/client-go/transport/websocket/roundtripper.go

```diff
@@ -111,6 +113,14 @@ func (rt *RoundTripper) RoundTrip(request *http.Request) (retResp *http.Response
 	wsConn, resp, err := dialer.DialContext(request.Context(), request.URL.String(), request.Header)
 	if err != nil {
 		if errors.Is(err, gwebsocket.ErrBadHandshake) {
+			// Enhance the error message with the http.Response data if possible.
+			if resp != nil {
+				errInfo := bytes.NewBufferString(resp.Status)
+				if errorBody, readErr := io.ReadAll(resp.Body); readErr == nil && len(errorBody) != 0 {
```

Comment:
I'm really unsure of using the response body from a dial / handshake error ... do we have precedent for this?

Reply:
After searching for precedence, I found none. So I removed adding the response body to the enhanced error. Please let me know what you think.